domain/machine/state/state.go

```diff
@@ -279,13 +318,19 @@ DELETE FROM net_node WHERE uuid IN
 			return errors.Errorf("deleting block devices for machine %q: %w", mName, err)
 		}
 
-		// Remove the status for the machine. No need to return error if no
+		// Remove the statuses for the machine. No need to return error if no
 		// status is set for the machine while deleting.
-		if err := tx.Query(ctx, deleteStatusStmt, machineUUIDParam).Run(); err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+		if err := tx.Query(ctx, deleteMachineStatusStmt, machineUUIDParam).Run(); err != nil {
 			return errors.Errorf("deleting status for machine %q: %w", mName, err)
 		}
+		if err := tx.Query(ctx, deleteMachineInstanceStatusStmt, machineUUIDParam).Run(); err != nil {
+			return errors.Errorf("deleting status for machine instance %q: %w", mName, err)
+		}
 
-		// Remove the machine.
+		// Remove the machine and instance rows.
+		if err := tx.Query(ctx, deleteMachineInstanceStmt, machineUUIDParam).Run(); err != nil {
+			return errors.Errorf("deleting machine %q: %w", mName, err)
```

Comment:
s/deleting machine/deleting machine instance/

To differentiate between the machine and the machine instance data.